packages/kbn-docs-utils/src/api_docs/stats.ts

```diff
@@ -59,7 +79,7 @@ function collectStatsForApi(doc: ApiDeclaration, stats: ApiStats, pluginApi: Plu
       collectStatsForApi(child, stats, pluginApi);
     });
   }
-  if (!doc.references || doc.references.length === 0) {
+  if (doc.references?.length === 0) {
```

Comment:
Removed the not having array check because that's essentially all the APIs that are not deprecated or not tracking adoption (unless the option captureRefererences is true).

Reply:
I'm not understanding, sorry, was this a bug before? `doc.references === undefined` feels like that would still qualify.

Reply:
Sorry... I should have been more explicit in my comment:

`references` are populated here: https://github.com/elastic/kibana/blob/92a8636f0ff63ab072527574e96e6616327b2ea4/packages/kbn-docs-utils/src/api_docs/build_api_declarations/build_basic_api_declaration.ts#L55

`maybeCollectReferences` only returns an array (empty or not) if the API's reference is findable and either the API is deprecated, is tracking adoption, or the `captureReferences` option is `true` (`false` by default).

IMO, checking that `doc.references === undefined` is a bug because, by default, it calls every API other than the deprecated (or tracking adoption) ones, as _unreferenced_.

FWIW, this array is only used in a place that depends on the `captureReferences` flag to be `true`: https://github.com/elastic/kibana/blob/8c94def73e04e48855d1d7f6b5252d341cdb98dd/packages/kbn-docs-utils/src/api_docs/build_api_docs_cli.ts#L181-L200

So I'm happy to revert this change if you think it's worth revisiting it. Maybe changing the `if` clause to the suggestion below?

```suggestion
  if (captureReferences && (!doc.references || doc.references?.length === 0)) {
```

Reply:
Honestly, I'm not sure how to answer here. I'm going to rely on your best judgement about how to proceed.

Reply:
FWIW, I'm not entirely sure either. The proposed implementation wouldn't mark as unreferenced the APIs that the compiler cannot find its references (`Node.isReferenceFindable() === false`). So we'd miss issues like #138511.

I'll revert the changes for now and I will create an issue to discuss it further.

Reply:
I created issue #139997 so we can discuss this.